test/ng/directive/formSpec.js

```diff
@@ -58,6 +58,78 @@ describe('form', function() {
     expect(form.alias).toBeUndefined();
   });
 
+
+  it('should ignore changes in manually removed controls', function() {
+    doc = $compile(
+      '<form name="myForm">' +
+        '<input name="control" ng-maxlength="1" ng-model="value" store-model-ctrl/>' +
+      '</form>')(scope);
+
+    var form = scope.myForm;
+
+    var input = doc.find('input').eq(0);
+    var inputController = input.controller('ngModel');
+
+    changeInputValue(input, 'ab');
+    scope.$apply();
+
+    expect(form.$error.maxlength).toBeTruthy();
+    expect(form.$dirty).toBe(true);
+    expect(form.$error.maxlength[0].$name).toBe('control');
+
+    // remove control
+    form.$removeControl(form.control);
+    expect(form.control).toBeUndefined();
+    expect(form.$error.maxlength).toBeFalsy();
+
+    inputController.$setPristine();
+    expect(form.$dirty).toBe(true);
+
+    form.$setPristine();
+
+    changeInputValue(input, 'abc');
+    scope.$apply();
+
+    expect(form.$error.maxlength).toBeFalsy();
+    expect(form.$dirty).toBe(false);
+  });
+
+
+  it('should react to changes in manually added controls', function() {
+    doc = $compile(
+      '<form name="myForm">' +
+        '<input name="control" ng-maxlength="1" ng-model="value" store-model-ctrl/>' +
+      '</form>')(scope);
+
+    var form = scope.myForm;
+
+    var input = doc.find('input').eq(0);
+    var inputController = input.controller('ngModel');
+
+    // remove control so we can add it again
+    form.$removeControl(form.control);
+    expect(form.control).toBeUndefined();
+    expect(form.$error.maxlength).toBeFalsy();
+    expect(form.$dirty).toBe(false);
```

Comment:
I would move these 3 expectations at line 119 (after the input has been edited, but before it has been added to the form), in order to verify that the form has been affected **only** after the control has been added back).
(It's not a big deal, since the previous test already verifies that a removed control does not affect the form, but it still makes me feel better :smiley:)